examples/run/cuda/full_chain_algorithm.hpp

```diff
@@ -47,8 +65,12 @@ class full_chain_algorithm
         const cell_container_types::host& cells) const override;
 
     private:
+    /// Host memory resource
+    std::reference_wrapper<vecmem::memory_resource> m_host_mr;
```

Comment:
Do we ever need to reseat this object such that this needs to be a reference wrapper instead of a reference?

Reply:
In the current setup we don't need it (yet), but an assignment operator may come in handy for this type later on. I don't feel super strongly about it though, so if you do, I might as well make it into a bare reference for now.

Reply:
After some further pondering I decided to just give in. The code should be a bit more elegant with a simple reference for now.